Lib/imaplib.py

```diff
@@ -1279,6 +1359,209 @@ def print_log(self):
                 n -= 1
 
 
+class Idler:
```

Comment:
If the class name is not supposed to be used, even in _isinstance_ checks, why not give it an underscore prefix?

```suggestion
class _Idler:
```

Reply:
I originally gave it an underscore prefix, but was asked to remove the prefix in an earlier review:

https://github.com/python/cpython/pull/122542#discussion_r1769601319

https://github.com/python/cpython/pull/122542#discussion_r1769601578

Should I change it back now?

Reply:
I don’t know. I didn’t realize this was contentious.

Reply:
In that case, I'm inclined to leave it as-is.

I originally used an underscore to signal that it's an internal interface, but that purpose is now served by the documentation, so restoring it would no longer add significant value. Moreover, it would require at least another documentation revision, and would undo something that was specifically requested by another reviewer before they gave their approval. I would rather not go down that road, as it would drag this review process out even longer than it has already taken.

Reply:
Marking resolved for now. If the underscore is important to you, and if it won't leave me stuck between two reviewers wanting contradictory changes, I would be happy to restore it. In that case, please re-open this and let me know.